test/unit/cursor/run_command_cursor.test.ts

```diff
@@ -0,0 +1,91 @@
+import { expect } from 'chai';
+
+import { MongoAPIError, MongoClient, RunCommandCursor } from '../../mongodb';
+
+describe('class RunCommandCursor', () => {
+  let client: MongoClient;
+  beforeEach(async function () {
+    client = new MongoClient('mongodb://iLoveJavascript');
+  });
+
+  context('constructor()', () => {
+    it('freezes and stores the command on the cursor instance', () => {
+      const cursor = client.db().runCursorCommand({ a: 1 });
+      expect(cursor).to.have.property('command').that.is.frozen;
+    });
+
+    it('creates getMoreOptions property with no defaults', () => {
+      const cursor = client.db().runCursorCommand({ a: 1 });
+      expect(cursor).to.have.property('getMoreOptions').that.deep.equals({});
+    });
+  });
+
+  context('setComment()', () => {
+    it('stores the comment value in getMoreOptions', () => {
+      const cursor = client.db().runCursorCommand({ a: 1 });
+      cursor.setComment('iLoveJS');
+      expect(cursor).to.have.nested.property('getMoreOptions.comment', 'iLoveJS');
+    });
+  });
+
+  context('setMaxTimeMS()', () => {
+    it('stores the maxTimeMS value in getMoreOptions.maxAwaitTimeMS', () => {
+      const cursor = client.db().runCursorCommand({ a: 1 });
+      cursor.setMaxTimeMS(2);
+      expect(cursor).to.have.nested.property('getMoreOptions.maxAwaitTimeMS', 2);
+    });
+
+    it('does not throw on incorrect maxTimeMS type', () => {
```

Comment:
suggestion: -> "does not validate maxTimeMS type"

why is this the desired/required behavior?

Reply:
updated title

- We already have validation / filtering in command construction, so only typeof number will be sent

Additional hypothetical scenarios:
- If someone put the wrong value the server throws, and as a build your own command API we shouldn't intervene (future fractional batchSize?)
- Any of the BSON numeric types are permitted as long as they are positive and do not have a fractional component, we don't need further validation that works against that reality (typeof number blocks Int32/Long instances)